.github/workflows/rustdoc.yml

```
@@ -0,0 +1,34 @@
+name: rustdoc
+on:
+  push:
+    branches:
+      - main
+
+env:
+  CARGO_INCREMENTAL: 0
+  CARGO_NET_RETRY: 10
+  RUSTFLAGS: "-D warnings -W unreachable-pub"
+  RUSTUP_MAX_RETRIES: 10
+
+jobs:
+  rustdoc:
+    if: github.repository == 'Xithrius/twitch-tui'
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+
+    - name: Install Rust toolchain
+      run: rustup update --no-self-update stable
+
+    - name: Build Documentation
+      run: cargo doc --all --no-deps
+
+    - name: Deploy Docs
+      uses: peaceiris/actions-gh-pages@v3
+      with:
+        github_token: ${{ secrets.GITHUB_TOKEN }}
+        publish_branch: gh-pages
```

```suggestion
        publish_branch: main
```

Don't change this yet, but I need to think about how this should actually be approached.

Perhaps a `docs` folder will be created in the main branch.

---

Yup a docs folder will be needed on the main branch

---

@Xithrius should we change this to be the main branch? or keep it in a separate branch?
A separate branch seems cleaner to me tbh, what do you think?

---

I like having everything in one place, so let's do main. If it's decided later that this isn't the way to do this project, then the docs can be moved to a new branch.